src/lib/libvkdt.h

```diff
@@ -0,0 +1,23 @@
+#ifndef LIBVKDT_H
+#define LIBVKDT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif 
+
+#include "qvk/qvk.h"
+#include "pipe/graph.h"
+#include "pipe/graph-io.h"
+#include "pipe/graph-print.h"
+#include "pipe/graph-export.h"
+#include "pipe/global.h"
+#include "pipe/modules/api.h"
+#include "core/log.h"
+#include "core/version.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
+
```

Comment:
the whole file is just a convenience thing? i suppose the cplusplus nonsense is useful to avoid a lot of typing, and also there may be a minimal necessary set of headers, so i guess this is fine.

Reply:
nah for a convenience thing leave the cplusplus guards in. as to api.. this probably needs careful consideration and probably internal restructuring too. some of the graph modification stuff is explicitly exposed to modules in `src/pipe/modules/api.h`. 

theu `ui_callback` is just a special case for a ui element that consists only of a button and triggers some code in the module. conceptually it's no more generic than a slider attached to a parameter. these ui hooks are listed here:
https://github.com/hanatos/vkdt/blob/master/src/pipe/global.c#L191

and yes, graph management is probably about to become more complex with the high level collapsible blocks with multiple inputs/outputs..